### PR TITLE
Link with /MANIFEST:EMBED to avoid UNRESOLVED tests

### LINKED ISSUE #811: tests/libcxx: re.alg.search/no_update_pos.pass.cpp is UNRESOLVED
For some reason, this is specific to **x86** - it doesn't repro when I build/test x64.

I observe 2 configurations of "Unresolved Tests":

```
S:\GitHub\STL\out\build\x86>python tests\utils\stl-lit\stl-lit.py ..\..\..\llvm-project\libcxx\test\std\re\re.alg\re.alg.search -o test.log
[...]
stl-lit.py: S:\GitHub\STL\tests\utils\stl\test\format.py:225: warning: [WinError 740] The requested operation requires elevation
UNRESOLVED: libc++ :: std/re/re.alg/re.alg.search/no_update_pos.pass.cpp:1 (19 of 24)
[...]
stl-lit.py: S:\GitHub\STL\tests\utils\stl\test\format.py:225: warning: [WinError 740] The requested operation requires elevation
UNRESOLVED: libc++ :: std/re/re.alg/re.alg.search/no_update_pos.pass.cpp:0 (22 of 24)
[...]
********************
Unresolved Tests (2):
  libc++ :: std/re/re.alg/re.alg.search/no_update_pos.pass.cpp:0
  libc++ :: std/re/re.alg/re.alg.search/no_update_pos.pass.cpp:1


Testing Time: 19.23s
  Expected Passes  : 14
  Expected Failures:  8
  Unresolved Tests :  2

1 warning(s) in tests
```

The log file contains:

```json
    {
      "code": "UNRESOLVED",
      "elapsed": 4.137548446655273,
      "name": "libc++ :: std/re/re.alg/re.alg.search/no_update_pos.pass.cpp:0",
      "output": "Exception during script execution:\nTraceback (most recent call last):\n  File \"S:/GitHub/STL/llvm-project/llvm\\utils\\lit\\lit\\worker.py\", line 73, in _execute_test_handle_errors\n    result = test.config.test_format.execute(test, lit_config)\n  File \"S:\\GitHub\\STL\\tests\\utils\\stl\\test\\format.py\", line 167, in execute\n    result = self._execute(test, lit_config)\n  File \"S:\\GitHub\\STL\\tests\\utils\\stl\\test\\format.py\", line 226, in _execute\n    raise e\n  File \"S:\\GitHub\\STL\\tests\\utils\\stl\\test\\format.py\", line 208, in _execute\n    self.test_executor.run(step.cmd, step.work_dir,\n  File \"S:\\GitHub\\STL\\tests\\utils\\stl\\test\\executor.py\", line 63, in run\n    out, err, rc = executeCommand(cmd, cwd=work_dir, env=env)\n  File \"S:\\GitHub\\STL\\tests\\utils\\stl\\util.py\", line 95, in executeCommand\n    p = subprocess.Popen(command, cwd=cwd,\n  File \"C:\\Users\\stl\\AppData\\Local\\Programs\\Python\\Python38\\lib\\subprocess.py\", line 854, in __init__\n    self._execute_child(args, executable, preexec_fn, close_fds,\n  File \"C:\\Users\\stl\\AppData\\Local\\Programs\\Python\\Python38\\lib\\subprocess.py\", line 1307, in _execute_child\n    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,\nOSError: [WinError 740] The requested operation requires elevation\n\n"
    },
```

The unescaped output is:

```
Exception during script execution:\nTraceback (most recent call last):
  File "S:/GitHub/STL/llvm-project/llvm\utils\lit\lit\worker.py", line 73, in _execute_test_handle_errors
    result = test.config.test_format.execute(test, lit_config)
  File "S:\GitHub\STL\tests\utils\stl\test\format.py", line 167, in execute
    result = self._execute(test, lit_config)
  File "S:\GitHub\STL\tests\utils\stl\test\format.py", line 226, in _execute
    raise e
  File "S:\GitHub\STL\tests\utils\stl\test\format.py", line 208, in _execute
    self.test_executor.run(step.cmd, step.work_dir,
  File "S:\GitHub\STL\tests\utils\stl\test\executor.py", line 63, in run
    out, err, rc = executeCommand(cmd, cwd=work_dir, env=env)
  File "S:\GitHub\STL\tests\utils\stl\util.py", line 95, in executeCommand
    p = subprocess.Popen(command, cwd=cwd,
  File "C:\Users\stl\AppData\Local\Programs\Python\Python38\lib\subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\stl\AppData\Local\Programs\Python\Python38\lib\subprocess.py", line 1307, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,\nOSError: [WinError 740] The requested operation requires elevation
```

I suspect that Windows is emitting "The requested operation requires elevation" because `no_update_pos.pass.exe` is an executable name containing `update` as a substring, and Windows contains a heuristic that such executables may be installers requiring admin elevation. :grinning: (I've seen this heuristic activate for `patch` executables in the past.)

I am unsure how to solve this:
* Somehow change how the executable is run? (We don't want to grant it admin privileges, but maybe we can disable the heuristic?)
* Somehow change how the executable is built - I know that there's a way to disable the heuristic by building in a manifest, but that seems highly invasive, especially for `libcxx`.
* Just skip the test?
* Ask `libcxx`'s maintainers if they'd consider renaming the test; I suspect that even `updating` would be sufficient to bypass the heuristic, and `change`, `modify`, etc. certainly would. This would preserve test coverage and permanently avoid the problem.

### PR DESCRIPTION
**Pull request by StephanTLavavej**
Fixes #811.

Windows has a feature, [Installer Detection Technology](https://docs.microsoft.com/en-us/windows/security/identity-protection/user-account-control/how-user-account-control-works#installer-detection-technology), that detects 32-bit executables without UAC manifests, with certain heuristics (like containing `update` in their filenames). This affects libcxx's `re.alg.search/no_update_pos.pass.cpp`. When the test harness attempts to run this executable, it requests admin permissions; the test harness doesn't display a UI prompt, but says that the test was "unresolved".

@AlexGuteniev figured out how to fix this non-intrusively - we just need to pass `/MANIFEST:EMBED` to the linker. This tells Windows that the executable is aware of UAC and doesn't require admin permissions.

Both `cl` and `clang-cl` are affected, so the linker option should be specified in the common set of options. It's fine that this is also applied for x64. So, this PR does 3 things:

1. Centralize options in `tests/universal_prefix.lst`, used by `std`, `tr1`, and `libcxx`.
2. Add `PM_LINK="/MANIFEST:EMBED"` to the universal options.
3. Enable more warnings in `tr1`, previously enabled for `std` and `libcxx`. Surprisingly, this requires no test code changes.

This will make it easier to add universal options in the future, makes it easier to see how `std`, `tr1`, and `libcxx` differ, and makes it harder for them to accumulate unnecessary divergence. (Thanks @BillyONeal!)

### CODE REVIEW

**Comment by BillyONeal** [comment]
Even though tr1 and std don't need it we might want to do it for consistency (since we want the same prefix.lst shared between them where possible)

**Comment by StephanTLavavej** [comment]
Good point. Perhaps we should centralize universal compiler/linker options across the three test suites.

**Comment by cbezault** [comment]
I'm kind of glad I optimistically implemented the `PM_LINK` behavior now.

File: tests/libcxx/usual_matrix.lst [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Copyright (c) Microsoft Corporation.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-PM_CL="/nologo /Od /W4 /w14061 /w14242 /w14265 /w14582 /w14583 /w14587 /w14588 /wd4643 /w14749 /w14841 /w14842 /w15038 /w15214 /w15215 /w15216 /w15217 /sdl /WX /Zc:strictStrings /D_ENABLE_STL_INTERNAL_CHECK /bigobj /FImsvc_stdlib_force_include.h /permissive-"
+RUNALL_INCLUDE ..\universal_prefix.lst
+RUNALL_CROSSLIST
+PM_CL="/wd4643 /FImsvc_stdlib_force_include.h /permissive-"
 RUNALL_CROSSLIST
 PM_CL="/EHsc /MTd /D_ITERATOR_DEBUG_LEVEL=2 /std:c++latest /analyze"
 PM_COMPILER="clang-cl" PM_CL="-fno-ms-compatibility -fno-delayed-template-parsing /EHsc /MTd /std:c++latest"
```
Comment (CaseyCarter) [inline tests/libcxx/usual_matrix.lst:9]: (1) There's quite a bit of overlap in `PM_CL` in the third section that could be extracted into the second, and (2) I find it peculiar that we explicitly specify `_ITERATOR_DEBUG_LEVEL` only for the cl config:
```suggestion
PM_CL="/wd4643 /FImsvc_stdlib_force_include.h /permissive- /EHsc /MTd /std:c++latest /D_ITERATOR_DEBUG_LEVEL=2"
RUNALL_CROSSLIST
PM_CL="/analyze"
PM_COMPILER="clang-cl" PM_CL="-fno-ms-compatibility -fno-delayed-template-parsing"
```

Comment (StephanTLavavej) [reply tests/libcxx/usual_matrix.lst:9]: Agreed - the explicit IDL=2 was a relic of being derived from the `std` configurations, but it's unnecessary here.

Comment (StephanTLavavej) [reply tests/libcxx/usual_matrix.lst]: Done - I dropped the unnecessary IDL=2 and rearranged the options slightly.

**Comment by StephanTLavavej** [comment]
Thanks for finding the right linker option, @AlexGuteniev!

**Pull request merged**